internal/k8s/builder/gateway.go

```diff
@@ -249,8 +252,13 @@ func (b *GatewayDeploymentBuilder) podSpec() corev1.PodSpec {
 						},
 					},
 				},
+				{
+					Name:  api.HTTPCAFile,
+					Value: consulCALocalFile,
```

Comment:
This adds the path to the CA cert in the pod's environment. This was previously done inside the shell script below as
```shell
export CONSUL_CACERT={{ .ConsulCAFile }}
```